mlflow/utils/_spark_utils.py

```diff
@@ -150,3 +151,24 @@ def get_or_extract(archive_path):
 
         _SparkDirectoryDistributor._extracted_dir_paths[archive_path] = temp_dir
         return _SparkDirectoryDistributor._extracted_dir_paths[archive_path]
+
+
+@contextlib.contextmanager
+def modified_environ(update):
+    """
+    Temporarily updates the ``os.environ`` dictionary in-place.
+
+    The ``os.environ`` dictionary is updated in-place so that the modification
+    is sure to work in all situations.
+
+    :param update: Dictionary of environment variables and values to add/update.
+    """
+    update = update or {}
+    original_env = {k: os.environ.get(k) for k in update}
+
+    try:
+        os.environ.update(update)
+        yield
+    finally:
+        for k, v in original_env.items():
+            os.environ.pop(k) if v is None else os.environ.update({k: v})
```

Comment:
```suggestion
            if v is None:
                os.environ.pop(k, None)
            else:
                os.environ.update({k: v})
```

Reply:
`os.environ.pop(k) if v is None else os.environ.update({k: v})` looks like we forget to return the result.